### PR TITLE
tsp, migrate tcgc getHttpOperationExamples

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**

### CODE REVIEW

File: typespec-extension/src/operation-utils.ts
Comment (weidongxu-microsoft) [inline typespec-extension/src/operation-utils.ts:1]: All change in this file is deleting code that no longer needed.

File: typespec-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -706,22 +704,19 @@ export class CodeModelBuilder {
...
+        operationExample["x-ms-original-file"] = pathToFileURL(example.filePath).toString();
+        operationExamples[operationExample.title ?? operationExample.operationId ?? operation.operation.name] = operationExample;
       }
-      operationExample = this.operationExamples.get(operation);
+      return operationExamples;
+    } else {
+      return undefined;
     }
-    return operationExample;
   }
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:720]: Most logic is same during the change.

One change is to make it directly return a Record that can be plugged in "x-ms-examples"

**Review by XiaofeiCao — APPROVED** [review]

File: typespec-extension/src/code-model-builder.ts [hunk 2]
```diff
@@ -706,22 +704,20 @@ export class CodeModelBuilder {
...
-      if (operation.projectionSource?.kind === "Operation") {
-        operation = operation.projectionSource;
+  private getOperationExample(operation: HttpOperation): Record<string, any> | undefined {
+    const httpOperationExamples = getHttpOperationExamples(this.sdkContext, operation);
+    if (httpOperationExamples && httpOperationExamples.length > 0) {
+      const operationExamples: Record<string, any> = {};
+      for (const example of httpOperationExamples) {
+        const operationExample = example.rawExample;
+        operationExample["x-ms-original-file"] = pathToFileURL(example.filePath).toString();
+        operationExamples[operationExample.title ?? operationExample.operationId ?? operation.operation.name] =
```
Comment (XiaofeiCao) [inline typespec-extension/src/code-model-builder.ts:714]: maybe the `operation.operation.name` can be removed, as TCGC will ensure either `title` or `operationId`:
https://github.com/Azure/typespec-azure/pull/1076/files#diff-e186052723c0e3dfe654f3a7fd0f36bd8126c371ad017fdee5a8643ba3d8da10R87

Comment (weidongxu-microsoft) [reply typespec-extension/src/code-model-builder.ts:714]: The rawExample is type `any`, so we don't have any constraint on type of this behavior.

Though I am actually relying on their logic on duplicate title.
https://github.com/Azure/typespec-azure/pull/1076/files#diff-d276532a6e7d2da177119679a7dfdc92a59980b7bbb70000f99ad0e4d50f3183R180

**Pull request merged**